Update to jQuery Foundation license

dmethvin opened the pull request:
This should all be good to go now.

dmethvin commented:
Oh sorry, i can fix that or you can rebase.

jdalton commented:
> Oh sorry, i can fix that or you can rebase.

If you rebase + force update your branch it'll fix it here.

jdalton commented on LICENSE:
```diff
@@ -1,23 +1,47 @@
-The MIT License (MIT)
+Copyright jQuery Foundation and other contributors, https://jquery.org/
 
```
No copyright years, nice! (one less thing to bump)

jdalton commented on LICENSE:
```diff
@@ -1,23 +1,47 @@
-The MIT License (MIT)
```
I notice this doesn't explicitly call out `MIT License`. Is there a place for that. I've had folks in the past not know off hand that the license is MIT without the little blurb.

dmethvin replied:
Since there was some extra wording added I wasn't sure whether we could keep this just saying it's the standard MIT license text. It's the same clauses of course but It's not a straight-text copy of the MIT license. I'll pull in @stevenayr for an opinion here. Note that the code file still does say MIT License.

dmethvin commented on LICENSE:
```diff
@@ -1,23 +1,47 @@
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+====
```
BTW @jdalton this is the straight boilerplate text for sample code that allows people to just copy/paste from a code sample rather than include a copyright. If you prefer MIT for that or there's no code samples in this repo we could remove it.

jdalton replied:
There's code samples in the docs so that clause is fine.

jdalton commented:
To fix the rebase issue you can do:
```bash
git checkout jqf-license
git pull -f https://github.com/lodash/lodash
git reset --hard d682357189fc8c15a1a3b598eb362bc4636903e9
git cherry-pick 4580fabc59b348bb3cc34c69c57afda8e3bf99bd
git push origin +jqf-license
```

dmethvin commented:
I fixed the commit and added a `license` field to the package.json as well which would be another place people might look for it. If it looks good to you it's ready to land.

jdalton commented:
Rock!

jdalton commented:
Don't worry about the other repos in the lodash or lodash-archive orgs. I'll copypasta this evening.
